codex/stores/networkstore.nim

```diff
@@ -108,6 +108,23 @@ method ensureExpiry*(
 
   return success()
 
+method ensureExpiry*(
+    self: NetworkStore,
+    treeCid: Cid,
+    index: Natural,
+    expiry: SecondsSince1970
+): Future[?!void] {.async.} =
+  ## Ensure that block's assosicated expiry is at least given timestamp
+  ## If the current expiry is lower then it is updated to the given one, otherwise it is left intact
+  ##
+
+  if (await self.localStore.hasBlock(treeCid, index)).tryGet:
```

Comment:
`tryGet` is dangerous here, it will raise an exception, instead of returning a failed result.

As a general rule, I'd avoid `get`, `tryGet`, `!`, etc. except in tests.